src/app/shared/validators/date/date.validator.ts

```diff
@@ -73,6 +73,28 @@ export function dateMonthYearValidator(dateType: string) {
   }
 }
 
+export function endDateMonthYearValidator() {
+  return (c: AbstractControl): { [key: string]: any } | null => {
+    const endDateYear = c.get('endDateGroup.endDateYear').value
+    const endDateMonth = c.get('endDateGroup.endDateMonth').value
+
+    const startDateYear = c.get('startDateGroup.startDateYear').value
+    const startDateMonth = c.get('startDateGroup.startDateMonth').value
+
+    if (!endDateYear && !startDateYear) {
```

Comment:
It seems like the users on the current UI can set only the start or the end date, it seems to me that this code assumes both days are always set. 

```suggestion
    if (!endDateYear || !startDateYear) {
```

Maybe I am reading something wrong? Please verify